internal/legacy/pinhal.go

```diff
@@ -0,0 +1,75 @@
+package legacy
+
+import "errors"
+
+// PinOutput represents a pin hardware abstraction layer for a pin that can output a digital signal.
+// This is an alternative to drivers.PinOutput abstraction which is a function type. Pros and cons
+// of both approaches have been discussed in the [relevant issue]. PinOutput should only be used
+// to expose an initialization function of a driver that receives pins of this type. Ideally
+// driver developers should also expose the initialization with drivers.Pin type:
+//
+//	func New(p1, p2, p3 legacy.PinOutput) *Device {
+//		return NewWithPinfuncs(p1.Set, p2.Set, p3.Set)
+//	}
+//
+//	func NewWithPinfuncs(p1, p2, p3 drivers.PinOutput) *Device {
+//		return &Device{p1:p1, p2:p2, p3:p3}
+//	}
+//
+// [relevant issue]: https://github.com/tinygo-org/drivers/pull/749/files
+type PinOutput interface {
+	Set(level bool)
+}
+
+// PinInput represents a pin hardware abstraction layer. See [PinOutput] for
+// more information on why this is "legacy".
+type PinInput interface {
```

Comment:
```suggestion
type PinGeter interface {
```
same reasoning as for the other interface name

Reply:
Same comment, using main thread for discussion https://github.com/tinygo-org/drivers/pull/753#issuecomment-3298195326